evm_loader/program/src/precompile_contracts.rs

```diff
@@ -279,6 +282,73 @@ pub fn erc20_wrapper<'a, B: AccountStorage>(
     }
 }
 
+// QueryAccount method ids:
+//------------------------------------------
+// owner(uint256)              => 0xa123c33e
+// length(uint256)             => 0xaa8b99d2
+// data(uint256,uint64,uint64) => 0x43ca5161
+//------------------------------------------
+
+const QUERY_ACCOUNT_METHOD_OWNER_ID: &[u8; 4] = &[0xa1, 0x23, 0xc3, 0x3e];
+const QUERY_ACCOUNT_METHOD_LENGTH_ID: &[u8; 4] = &[0xaa, 0x8b, 0x99, 0xd2];
+const QUERY_ACCOUNT_METHOD_DATA_ID: &[u8; 4] = &[0x43, 0xca, 0x51, 0x61];
+
+/// Call inner `query_account`
+#[must_use]
+pub fn query_account<'a, B: AccountStorage>(
+    input: &[u8],
+    state: &mut ExecutorState<'a, B>
+)
+    -> Capture<(ExitReason, Vec<u8>), Infallible>
+{
+    let (method_id, rest) = input.split_at(4);
+    let method_id: &[u8; 4] = method_id.try_into().unwrap_or_else(|_| &[0_u8; 4]);
+    let (account_address, rest) = rest.split_at(32);
+    let account_address = Pubkey::new(account_address);
+
+    match method_id {
+        QUERY_ACCOUNT_METHOD_OWNER_ID => {
+            debug_print!("query_account get owner {}", account_address);
+            let owner = state.query_solana_account_owner(account_address);
+            if let Some(owner) = owner {
+                debug_print!("query_account owner result: {:?}", owner);
+                return Capture::Exit((ExitReason::Succeed(evm::ExitSucceed::Returned), owner.as_ref().to_owned()));
+            }
+            let revert_message = b"QueryAccount owner failed".to_vec();
+            Capture::Exit((ExitReason::Revert(evm::ExitRevert::Reverted), revert_message))
+        },
+        QUERY_ACCOUNT_METHOD_LENGTH_ID => {
+            debug_print!("query_account get length {}", account_address);
+            let length = state.query_solana_account_length(account_address);
+            if let Some(length) = length {
+                debug_print!("query_account length result: {:?}", length);
+                let length = &length.to_be_bytes(); // pad to 32
```

Comment:
Do what comment says.
Convert `length` to `U256` before serializing it to vector

Reply:
Done